packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -479,25 +479,31 @@ async function _getAvailableSlots({ input, ctx }: GetScheduleOptions): Promise<I
     throw new TRPCError({ message: "Invalid time range given.", code: "BAD_REQUEST" });
   }
 
-  const eventHosts = await monitorCallbackAsync(
-    findQualifiedHosts<GetAvailabilityUser>,
-    eventType,
-    !!input.rescheduleUid
-  );
-  const hostsAfterSegmentMatching = await findMatchingHostsWithEventSegment({
-    eventType,
-    normalizedHosts: eventHosts,
-  });
   const contactOwnerEmailFromInput = input.teamMemberEmail ?? null;
   const skipContactOwner = input.skipContactOwner;
   const contactOwnerEmail = skipContactOwner ? null : contactOwnerEmailFromInput;
   const routedTeamMemberIds = input.routedTeamMemberIds ?? null;
+
   const routedHostsWithContactOwnerAndFixedHosts = getRoutedHostsWithContactOwnerAndFixedHosts({
```

Comment:
Changing the order around, ensuring that findQualifiedHosts runs after routing form routed hosts - otherwise it's possible that findQualifiedHosts excludes all users routed to in the routing form.